src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1384,77 +1382,66 @@ Literal &Literal::operator+=(const Literal &other) noexcept {
 }
 
 Literal Literal::sub(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
-    // TODO subtype handling
-
     // DateTime(&Subclasses) - DateTime(&Subclasses)
-    if (this->is_timepoint_like() && other.is_timepoint_like() && this->datatype_id() == other.datatype_id()) {
```

Comment:
If `is_timepoint_like` and `is_duration_like` are no longer used you can remove them again.